src/components/util/getBarChartOptions.ts

```diff
@@ -130,46 +165,97 @@ export default function getBarChartOptions({
                 label += '%';
               }
             }
             return label;
-          }
-        }
+          },
+        },
       },
       datalabels: {
-        //https://chartjs-plugin-datalabels.netlify.app/guide/
-        anchor: stacked || stackMetrics ? 'center' : 'end',
-        align: stacked || stackMetrics ? 'center' : 'end',
-        display: showLabels ? 'auto' : false,
-        formatter: (v, context) => {
-          //metric needed for formatting
-          const metricIndex = context.datasetIndex;
-          //a single metric is sometimes passed in (e.g. for stacked bar charts)
-          const metricsList = [...(metrics || []), ...(lineMetrics||[])];
-          const metricObj = metrics ? metricsList[metricIndex] : metric;
-          if (v === null) return null;
-          let val = formatValue(v, { 
-            type: 'number', 
-            dps: dps,
-            meta: displayAsPercentage ? undefined : metricObj?.meta 
-          });
-          if (displayAsPercentage) {
-            val += '%';
-          }
-          return val;
-        }
-      }
-    }
+        labels: {
+          total: {
+            anchor: (context) => {
+              const dataset = context.dataset as ExtendedChartDataset;
+              const totals = dataset.totals;
+              if (!totals) {
+                return 'end';
+              }
+              const currXAxisName = dataset.xAxisNames?.[context.dataIndex];
+              const currTotal = totals[currXAxisName || '']?.total;
+              if (currTotal && currTotal < 0) {
+                return 'start';
+              }
+              return 'end';
+            },
+            align: (context) => {
+              const dataset = context.dataset as ExtendedChartDataset;
+              const totals = dataset.totals;
+              if (!totals) {
+                return displayHorizontally ? 'right' : 'top';
+              }
+              const currXAxisName = dataset.xAxisNames?.[context.dataIndex];
+              const currTotal = totals[currXAxisName || '']?.total;
+              if (currTotal && currTotal < 0) {
+                return displayHorizontally ? 'left' : 'bottom';
+              }
+              return displayHorizontally ? 'right' : 'top';
+            },
+            display: showTotals ? 'true' : false,
+            font: {
+              weight: 'bold',
+            },
+            formatter: (v, context) => {
+              const dataset = context.dataset as ExtendedChartDataset;
+              const xAxisNames = dataset.xAxisNames;
+              const totals = dataset.totals;
+              if (!totals || !xAxisNames) {
+                return '';
+              }
+              const currxAxisName = xAxisNames[context.dataIndex];
+              const currDatasetIndex = context.datasetIndex;
+              if (currDatasetIndex === totals[currxAxisName].lastSegment && v !== null) {
+                return totals[currxAxisName].total;
```

Comment:
@cwbuecheler I'm wondering here if the user would want the total formatted like the values (see below)? i.e. include any post + pretext, and add commas etc to large numbers

Reply:
Good call, will add that in